_playwright-tests/UI/CustomRepoPagination.spec.ts

```diff
@@ -0,0 +1,49 @@
+import { test, expect } from '@playwright/test';
+import { navigateToRepositories } from './helpers/navHelpers';
+import { deleteAllRepos } from './helpers/deleteRepositories';
+import { bulkCreateRepos } from './helpers/createRepositories';
+import { closePopupsIfExist } from './helpers/helpers';
+
+const repoNamePrefix = 'custom_repo-pagination';
+
+test.describe('Custom repositories pagination', () => {
+  test('Test pagination on custom repo page', async ({ page }) => {
+    await test.step('Delete any pagination test repos that exist', async () => {
+      await deleteAllRepos(page, `&search=${repoNamePrefix}`);
+      await page.waitForTimeout(1500); // Wait for soft delete
```

Comment:
This is `sleep`. It will be flaky sooner or later - either it waits too long unnecessarily or it waits for too short. If we need to wait for something not to be present, we should wait in different manner. Also, doesn't `deleteAllRepos` function waits until it finishes?

Reply:
Hi, I have removed that now. We had some URL collisions and tought it might be due soft delete taking too long after the API call to delete. It only happened when  repeatedly running the test.